### PR TITLE
Region specific settings

### PR DESCRIPTION
**Pull request by crast**

### CODE REVIEW

File: src/app/Api/Context.php [hunk 1]
```diff
@@ -127,33 +128,60 @@ public function setRegion(Models\Region $region)
...
+                $center = $regionOrCenter;
+                $region = $center->region;
+            }
         }
-        if ($setting != null) {
-            return $setting->value;
+
+        $inputs = compact('center', 'quarter', 'region');
+
+        foreach (static::$settingsLookups as $config) {
```
Comment (pdarg) [inline src/app/Api/Context.php]: can you add some comments for this. it's pretty meta

File: src/app/Traits/ScopedSettings.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $rawSetting = $this->scopedSettingBaseQuery()->name($setting)->first();
+
+        return ($rawSetting !== null) ? $rawSetting->value : null;
+
+    }
+
+    public function getScopedSetting(string $setting)
+    {
+        if (++static::$_fetches < 2) {
```
Comment (pdarg) [inline src/app/Traits/ScopedSettings.php]: this is never incremented. is that intentional?

Comment (crast) [inline src/app/Traits/ScopedSettings.php]: that is an increment command right there ^

Comment (pdarg) [inline src/app/Traits/ScopedSettings.php]: doh

Comment (crast) [inline src/app/Traits/ScopedSettings.php]: I will add a comment however on what it's actually doing.

**Pull request merged**